test/reduce-e2e/reduce_test.go

```diff
@@ -229,8 +229,8 @@ func (r *ReduceSuite) TestComplexSlidingWindowPipeline() {
 	}()
 
 	w.Expect().
-		SinkContains("sink", "15").
-		SinkContains("sink", "45").
+		SinkContains("sink", "30").
```

Comment:
@yhl25 why will this change? because we change the watermark position?

Reply:
yes that explains why this would change.

Reply:
this pipeline has 5 second fixed keyed window, 10s non keyed fixed followed by 60s sliding with 10s slide. 
so every 10s, we should see 30 being emitted, but by the earlier logic, it was emitting wrong values because of the slide.

Reply:
yes because the event time in the message is changed to the start time of the partition.